### PR TITLE
[LLM Runtime] Add Whisper Example and Python API

### PR DESCRIPTION
**Pull request by a32543254**
## Type of Change

feature or bug fix or documentation or others
API changed or not

## Description

- [x] Update whisper Cmakelist
- [x] Fp32/Q4_0 inference
- [x] README update
- [ ] Jblas and MHA fusion + FFN fusion
- [x] Python API 
- [x] Audio Quant 
- [ ] Extension test

## Expected Behavior & Potential Risk

the expected behavior that triggered by this PR 

## How has this PR been tested?

how to reproduce the test (including hardware information)

## Dependency Change?

any library dependency introduced or removed

### CODE REVIEW

**Comment by airMeng** [comment]
Please add auto-formatting by clang-format to your IDE auto-save options.

File: intel_extension_for_transformers/llm/runtime/graph/README.md [hunk 1]
```diff
@@ -39,7 +39,7 @@ LLM Runtime supports the following models:
 |[Baichuan-13B-Chat](https://huggingface.co/baichuan-inc/Baichuan-13B-Chat), [Baichuan2-13B-Chat](https://huggingface.co/baichuan-inc/Baichuan2-13B-Chat)| ✅ | ✅ |  Latest |
 |[Mistral-7B](https://huggingface.co/mistralai/Mistral-7B-v0.1)| ✅ | ✅ | 4.34.0 or newer |
 |[Qwen-7B](https://huggingface.co/Qwen/Qwen-7B-Chat), [Qwen-14b](https://huggingface.co/Qwen/Qwen-14B-Chat)| ✅ | ✅ |  Latest |
-
+|[Whisper-Tiny](https://huggingface.co/openai/whisper-tiny), [Whisper-Base](https://huggingface.co/openai/whisper-base), [Whisper-Small](https://huggingface.co/openai/whisper-small), [Whisper-Medium](https://huggingface.co/openai/whisper-medium),[Whisper-Large](https://huggingface.co/openai/whisper-large)| ✅ | ✅ |  Latest |
```
Comment (hshen14) [inline intel_extension_for_transformers/llm/runtime/graph/README.md:42]: any code sample to show how to run whisper?

Comment (a32543254) [reply intel_extension_for_transformers/llm/runtime/graph/README.md:42]: After python api for whisper done.
the sample code should be the same as current api example code.
Only need replace text input to audio input.

Comment (hshen14) [reply intel_extension_for_transformers/llm/runtime/graph/README.md:42]: OK, can you add an example in readme? We should already have audio sample in our assets.

**Comment by intellinjun** [comment]
image
whisper q40 output

File: intel_extension_for_transformers/llm/runtime/graph/__init__.py [hunk 2]
```diff
@@ -141,6 +143,11 @@ def quant_model(self, model_type, model_path, out_path, **quant_kwargs):
         self.__import_package(model_type)
         self.module.Model.quant_model(model_path=model_path, out_path=out_path, **quant_kwargs)
 
+    def inference(self, audio_file):
```
Comment (zhentaoyu) [inline intel_extension_for_transformers/llm/runtime/graph/__init__.py:146]:
```suggestion
def __call__(self, input_ids, reinit=False, audio_file=None, **kwargs):
        if self.model is None:
            self.init_from_bin(self.model_type, self.bin_file, **kwargs)
            self.generate_round = 0
        elif reinit:
            self.model.reinit()
            self.generate_round = 0
        if input_ids is not None:
            return self.model.evaluate(input_ids.tolist())
         if audio_file is not None:
             return self.model.inference(audio_file)
```
we can also add `model_type` liked args in kwargs for different models inference with different types

File: intel_extension_for_transformers/llm/runtime/graph/__init__.py [hunk 3]
```diff
@@ -141,6 +143,11 @@ def quant_model(self, model_type, model_path, out_path, **quant_kwargs):
...
+        self.model = self.module.Model()
+        self.model.init_model(self.bin_file)
```
Comment (lzwhard) [inline intel_extension_for_transformers/llm/runtime/graph/__init__.py:148]: init once, multi-inference

**Review by zhenwei-intel — APPROVED** [review]

File: intel_extension_for_transformers/llm/runtime/graph/__init__.py (hunk 2 above)
Comment (zhentaoyu) [reply intel_extension_for_transformers/llm/runtime/graph/__init__.py]: [700bbc2](https://github.com/intel/intel-extension-for-transformers/pull/809/commits/700bbc2c4b71bfabbed14331da0eec6edcbbe841) looks good to me

**Review by zhentaoyu — APPROVED** [review]

**Comment by VincyZhang** [comment]
https://inteltf-jenk.sh.intel.com/view/nlp-toolkit-validation/job/ITREX-cpp-graph-extension/252/artifact/deploy/engine/whisper/launch.log

**Pull request merged**